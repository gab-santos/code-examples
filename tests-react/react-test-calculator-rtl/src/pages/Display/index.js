import React from "react";

import "./styles.css";

export default function Display({ displayValue }) {
  return (
    <div testid="display-container" className="display-container">
      <p testid="display-value" className="display-value">
        {displayValue}
      </p>
    </div>
  );
}
