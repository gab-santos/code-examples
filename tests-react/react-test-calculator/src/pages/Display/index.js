import React from "react";

import "./styles.css";

export default function Display({ displayValue, setDisplayValue }) {
  return (
    <div className="display-container">
      <p className="display-value">{displayValue}</p>
    </div>
  );
}
