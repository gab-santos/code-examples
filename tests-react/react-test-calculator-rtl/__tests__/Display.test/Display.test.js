import React from "react";
import { cleanup, render } from "@testing-library/react";
import Display from "../../src/pages/Display";

describe("Display", () => {
  beforeEach(() => {
    cleanup();
  });

  it("should render correctly", () => {
    const { container } = render(<Display />);
    expect(container).toMatchSnapshot();
  });

  it("should renders the value of displayValue", () => {
    const { getByTestId } = render(<Display displayValue={"test"} />);
    const displayValue = getByTestId("display-value");

    expect(displayValue).toBeInTheDocument();
    expect(displayValue.textContent).toBe("test");
  });
});
