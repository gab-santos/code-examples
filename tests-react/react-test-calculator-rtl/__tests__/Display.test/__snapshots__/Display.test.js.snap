// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display should render correctly 1`] = `
<div>
  <div
    class="display-container"
    testid="display-container"
  >
    <p
      class="display-value"
      testid="display-value"
    />
  </div>
</div>
`;
